src/app/Main/Body/QueryPage/QueryPage-Container.component.tsx

```diff
@@ -31,23 +32,32 @@ export function QueryPageContainerComponent(): JSX.Element {
     const fetchAndSetDataAndFilter = async (): Promise<void> => {
         const isolateResponse: Response = await fetch(ISOLATE_URL);
         const isolateProp: IsolateDTO = await isolateResponse.json();
-        const keyValueProps = Object.keys(isolateProp.isolates[0]) as DbKey[];
+
+        const adaptedDbIsolates: DbCollection = isolateProp.isolates.map(
+            ({ microorganism, samplingContext, matrix }) => ({
+                microorganism,
+                samplingContext,
+                matrix,
+            })
+        );

```

Comment:
Nice. Now if the api keys changes you just need to change this map function and the rest of your code works as before
Same for the filter props below